torch/csrc/jit/script/compiler.cpp

```diff
@@ -418,45 +421,57 @@ struct to_ir {
     }
   }
 
-  void emitWhile(const While& stmt) {
-    // Emits a loop operators conforming to the semantics specified at
-    // https://github.com/onnx/onnx/blob/master/docs/Operators.md#experimental-loop
-    // TODO: implement scan_outputs
-
-    // the format of the Loop instruction is:
-    // loop_carried_outputs* = Loop(max_trip_count, start_condition, loop_carried_inputs*)
-    //                          block0(loop_counter, loop_carried_block*) {
-    //                             <body>
-    //                             -> (continue_condition, loop_carried_block_outputs*)
-    //                          }
-    // all loop_carried_... lists are the same length and represent the value of
-    // loop-carried variables whose definitions are updated as the loop executes
-    // in a way that ensure single static assignment.
-
-    // TODO: clarify that this is an optional input that isn't needed here
-    Value* max_trip_count_dummy = emitConst(stmt.range(), INT_MAX, "i")[0];
-    Value* cond_value = emitExpr(stmt.cond(), 1)[0];
-
+  // *********************** Loop Operators ************************************
+  // Emits a loop operators conforming to the semantics specified at
+  // https://github.com/onnx/onnx/blob/master/docs/Operators.md#experimental-loop
+  // TODO: implement scan_outputs
+
+  // the format of the Loop instruction is:
+  // loop_carried_outputs* = Loop(max_trip_count, start_condition,
+  // loop_carried_inputs*)
+  //                          block0(loop_counter, loop_carried_block*) {
+  //                             <body>
+  //                             -> (continue_condition,
+  //                             loop_carried_block_outputs*)
+  //                          }
+  // all loop_carried_... lists are the same length and represent the value of
+  // loop-carried variables whose definitions are updated as the loop executes
+  // in a way that ensure single static assignment.
+
+  void emitLoopCommon(
+      Value* max_trip_count,
+      Value* cond_value,
+      const List<Stmt>& body,
+      const Stmt& stmt,
+      const Expr* cond,
+      const std::vector<std::string>& itr_idents) {
```

Comment:
It's not really itr_idents, it's `trip_count_ident`. You can just accept a single `at::optional` and put the asserts in `emitFor`

Reply:
This is meant to allow us to bind multiple iteration variables in the future, e.g. `for i, j, k in ...`. Right now we don't support that (should we? cc @zdevito) so this looks pretty stupid but that's the reasoning behind it

Reply:
I understand the reasoning, but if we're not going to support multiple iterators any time soon then it will only be more confusing and unnecessarily complicate the code. When are multiple iter variables important if you don't have tuples, lists, etc.? The only case that comes up to my mind is `enumerate`.

Reply:
Let's just have a single `Ident` here. We should use Ident rather than std::string since it carries more debug info.  It is easy to add multiple Idents when we know what we will do with them, but it is confusing to have it look like we are supporting them but then not doing anything with them.

Reply:
Also, can we take a single `Expr` rather than both `Value*` for the initial value and the `Expr` option for the next value? It is code smell to have a single compiler function take both pre- and post- translated values as inputs.